proxy_collection.go

```diff
@@ -38,6 +38,29 @@ func (collection *ProxyCollection) Add(proxy *Proxy, start bool) error {
 	return nil
 }
 
+func (collection *ProxyCollection) AddOrReplace(proxy *Proxy, start bool) error {
+	collection.Lock()
+	defer collection.Unlock()
+
+	if existing, exists := collection.proxies[proxy.Name]; exists {
+		if existing.Listen == proxy.Listen && existing.Upstream == proxy.Upstream {
```

Comment:
Yeah, doing the resolving sounds good. Can be done in a follow-up, but we should create an issue for it unless you're planning to address it right away.